src/main/java/com/shaft/tools/io/internal/ExecutionSummaryReport.java

```diff
@@ -50,39 +50,52 @@ public static void generateExecutionSummaryReport(int passed, int failed, int sk
-                        .replace("${SKIPPED_DROPDOWN_OPTION}", StatusIcon.SKIPPED.getValue() + Status.SKIPPED.name())
-                        .replace("${CASES_DETAILS}", detailsBuilder));
+                createReportMessage(passed, failed, skipped, startTime, endTime, detailsBuilder));
 
         ReportManagerHelper.openExecutionSummaryReportAfterExecution();
         ReportManagerHelper.logExecutionSummary(String.valueOf(total), String.valueOf(passed), String.valueOf(failed), String.valueOf(skipped));
     }
 
+    private static String createReportMessage(int passed, int failed, int skipped, long startTime, long endTime, StringBuilder detailsBuilder) {
+        int total = passed + failed + skipped;
+        var report = HTMLHelper.EXECUTION_SUMMARY.getValue()
+                .replace("${LOGO_URL}", SHAFT_LOGO_URL)
+                .replace("${DATE}", new SimpleDateFormat("dd/MM/yyyy").format(endTime))
+                .replace("${START_TIME}", new SimpleDateFormat("HH:mm:ss").format(startTime))
+                .replace("${END_TIME}", new SimpleDateFormat("HH:mm:ss").format(endTime))
+                .replace("${TOTAL_TIME}", ReportManagerHelper.getExecutionDuration(startTime, endTime))
+                .replace("${CASES_TOTAL}", String.valueOf(total))
+                .replace("${CASES_PASSED}", String.valueOf(passed))
+                .replace("${CASES_FAILED}", String.valueOf(failed))
+                .replace("${CASES_SKIPPED}", String.valueOf(skipped))
+                .replace("${VALIDATION_PASSED_PERCENTAGE_PIE}", String.valueOf(passedValidations * 360d / validations.size()))
+                .replace("${VALIDATION_PASSED_PERCENTAGE}", String.valueOf(new DecimalFormat("0.00").format((float) passedValidations * 100 / validations.size())))
+                .replace("${VALIDATION_TOTAL}", String.valueOf(validations.size()))
+                .replace("${VALIDATION_PASSED}", String.valueOf(passedValidations))
+                .replace("${VALIDATION_FAILED}", String.valueOf(failedValidations))
+                .replace("${TOTAL_ISSUES}", String.valueOf(ReportManagerHelper.getIssueCounter()))
+                .replace("${NO_OPEN_ISSUES_FAILED}", String.valueOf(ReportManagerHelper.getFailedTestsWithoutOpenIssuesCounter()))
+                .replace("${OPEN_ISSUES_PASSED}", String.valueOf(ReportManagerHelper.getOpenIssuesForPassedTestsCounter()))
+                .replace("${OPEN_ISSUES_FAILED}", String.valueOf(ReportManagerHelper.getOpenIssuesForFailedTestsCounters()))
+                .replace("${PASSED_DROPDOWN_OPTION}", StatusIcon.PASSED.getValue() + Status.PASSED.name())
+                .replace("${FAILED_DROPDOWN_OPTION}", StatusIcon.FAILED.getValue() + Status.FAILED.name())
+                .replace("${SKIPPED_DROPDOWN_OPTION}", StatusIcon.SKIPPED.getValue() + Status.SKIPPED.name())
+                .replace("${CASES_DETAILS}", detailsBuilder);
+
+        if (total > 0) {
```

Comment:
@MahmoudElSharkawy note that if you run no checkpoints (a test that only has steps without any verifications) the report will face "division by zero" and will output NaN error instead of the pie chart... it's a stupid issue, but I thought to fix it blmara.